### PR TITLE
Add a pools.yaml generator example

### PR DESCRIPTION
**Pull request by omersch381**
This patch adds an example to a pools.yaml file generator, it uses dummy data that was taken from a devstack deployment, and does not support multipool (yet).

### CODE REVIEW

File: demo/examples/generate_bind9_multipool_pools_yaml/generate_bind9_multipool_pools_yaml.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+				},
+			},
+		},
+		CatalogZone: catalogZone,
+	}
+	// Second pool definition
+	attributes = map[string]string{
+		"type": "external",
+	}
+	pool2 := Pool{
```
Comment (beagles) [inline demo/examples/generate_bind9_multipool_pools_yaml/generate_bind9_multipool_pools_yaml.go:110]: This sample data should include multiple entries for binds and mdns servers.

**Comment by omersch381** [comment]
The generated file I printed to the logs:
2024-10-04T15:08:31Z	INFO	Controllers.Designate	pools.yaml content is 
---
- name: default
  description: DevStack BIND Pool
  attributes: {
    "pool_level": "default",
    "type": "internal",
  }

  ns_records:
    - hostname: ns1.example.org.
      priority: 1
    - hostname: ns2.example.org.
      priority: 2
    - hostname: ns3.example.org.
      priority: 3

  nameservers:
    - host: 172.28.0.99
      port: 53
    - host: 172.28.0.98
      port: 53

  targets:
    - type: bind9
      description: BIND9 Server 1 (172.28.0.99)

      masters:
        - host: 172.28.0.97
          port: 16000
        - host: 172.28.0.97
          port: 16001
        - host: 172.28.0.97
          port: 16002

      options:
        host: 172.28.0.99
        port: 53
        rndc_host: 172.28.0.99
        rndc_port: 953
        rndc_config_file: /etc/designate/rndc-keys/rndc-key-1.conf
    - type: bind9
      description: BIND9 Server 2 (172.28.0.98)

      masters:
        - host: 172.28.0.97
          port: 16000
        - host: 172.28.0.97
          port: 16001
        - host: 172.28.0.97
          port: 16002

      options:
        host: 172.28.0.98
        port: 53
        rndc_host: 172.28.0.98
        rndc_port: 953
        rndc_config_file: /etc/designate/rndc-keys/rndc-key-2.conf

when I had those before (but even if I didn't, they would get created):
~ oc get configmaps designate-mdns-ip-map -o json | jq -r '.data'; oc get configmaps designate-bind-ip-map -o json | jq -r '.data'
{
  "mdns_crc-vlf7c-master-0": "172.28.0.97"
}
{
  "bind_address_0": "172.28.0.98",
  "bind_address_1": "172.28.0.99"
}

and that configmap, which was created in the current PR:
~ > oc get configmaps designate-pools-yaml-config-map -o json | jq '.data'                      
{
  "designate-pools-yaml-config-map": "\n---\n- name: default\n  description: DevStack BIND Pool\n  attributes: {\n    \"pool_level\": \"default\",\n    \"type\": \"internal\",\n  }\n\n  ns_records:\n    - hostname: ns1.example.org.\n      priority: 1\n    - hostname: ns2.example.org.\n      priority: 2\n    - hostname: ns3.example.org.\n      priority: 3\n\n  nameservers:\n    - host: 172.28.0.99\n      port: 53\n    - host: 172.28.0.98\n      port: 53\n\n  targets:\n    - type: bind9\n      description: BIND9 Server 1 (172.28.0.99)\n\n      masters:\n        - host: 172.28.0.97\n          port: 16000\n        - host: 172.28.0.97\n          port: 16001\n        - host: 172.28.0.97\n          port: 16002\n\n      options:\n        host: 172.28.0.99\n        port: 53\n        rndc_host: 172.28.0.99\n        rndc_port: 953\n        rndc_config_file: /etc/designate/rndc-keys/rndc-key-1.conf\n    - type: bind9\n      description: BIND9 Server 2 (172.28.0.98)\n\n      masters:\n        - host: 172.28.0.97\n          port: 16000\n        - host: 172.28.0.97\n          port: 16001\n        - host: 172.28.0.97\n          port: 16002\n\n      options:\n        host: 172.28.0.98\n        port: 53\n        rndc_host: 172.28.0.98\n        rndc_port: 953\n        rndc_config_file: /etc/designate/rndc-keys/rndc-key-2.conf\n"
}

File: demo/examples/generate_bind9_multipool_pools_yaml/generate_bind9_multipool_pools_yaml.go (hunk 1 above)
Comment (omersch381) [reply demo/examples/generate_bind9_multipool_pools_yaml/generate_bind9_multipool_pools_yaml.go:109]: Done

**Comment by omersch381** [comment]
Now the templating module supports multiple pools. I have just tested it on my host.

**Comment by omersch381** [comment]
This PR served as a "Spike" (a commit that is mainly used for researching and POC). I am closing it now and will start working on the implementation.

**Pull request closed without merging**